### PR TITLE
Improved compatibility for non-jQuery and non-Bootstrap projects

### PR DESCRIPTION
**Pull request by tbentham**
Hey,

I'm with https://github.com/geekzonehq and we forked this project a while back to make some improvements so it would be compatible with our own project, which doesn't use jQuery or Bootstrap.

The changes include removing all jQuery references outside of the `modal` function, and adding optional chaining to the `modal` function so the script still works even if Bootstrap isn't referenced.

It should work just the same with Bootstrap - I've tested it briefly - but feel free to test it yourself and let me know what you think.

Thanks!

### CODE REVIEW

File: cookiebanner/templates/cookiebanner/bootstrap4.html [hunk 1]
```diff
@@ -59,26 +59,25 @@ <h4 class="cookiebannerH4">{{ cookiegroup.name }}</h4>
...
-    $(".cookiebannerSubmit").click(function (e) {
+    Array.from(document.querySelectorAll(".cookiebannerSubmit")).forEach((btn) => btn.addEventListener("click", function () {
       let enable_cookies;
       if (this.name === 'enable_all') {
         enable_cookies = cookiegroups.map((x) => x.id);
       } else {
-        let serialized_form = $("#cookiebannerForm").serializeArray();
-        let checked_cookiegroups = serialized_form.map((x) => x.name);
+        let checked_cookiegroups = Array.from(document.querySelector("#cookiebannerForm"))
+          .filter((x) => x.checked).map((x) => x.name);
```
Comment (andreasnuesslein) [inline cookiebanner/templates/cookiebanner/bootstrap4.html:76]: @tbentham this is the only part that I'm not sure about. - Are you certain this works as expected?

File: cookiebanner/templates/cookiebanner/bootstrap4.html [hunk 2]
```diff
@@ -59,26 +59,25 @@ <h4 class="cookiebannerH4">{{ cookiegroup.name }}</h4>
...
 
   document.addEventListener("DOMContentLoaded", function () {
-    let keyValue = document.cookie.match('(^|;) ?cookiebanner=([^;]*)(;|$)');
-    let cookiebannerCookie = keyValue ? decodeURIComponent(keyValue[2]) : null;
+    let keyValue = document.cookie.split(/;\s*/g).find((x) => x.startsWith("cookiebanner"));
+    let cookiebannerCookie = keyValue ? decodeURIComponent(keyValue.slice(13)) : null;
     if (cookiebannerCookie) return;
 
-    $('#cookiebannerModal').modal({
+    $?.('#cookiebannerModal').modal?.({
```
Comment (andreasnuesslein) [inline cookiebanner/templates/cookiebanner/bootstrap4.html:66]: Will this work in older browsers? I'm thinking `$?.` might not be supported by older ones

**Comment by andreasnuesslein** [comment]
Thank you for this pull request! We're all quite swamped but hopefully we'll be able to merge this soon.

File: cookiebanner/templates/cookiebanner/bootstrap4.html (hunk 2 above)
Comment (tbentham) [reply cookiebanner/templates/cookiebanner/bootstrap4.html:66]: According to [this](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining#browser_compatibility), it should be compatible with all major browsers except Internet Explorer. However, it is a very recent feature.

I have seen another solution using `if (typeof foo.functionName === "function")`. Would something like that be preferable?

File: cookiebanner/templates/cookiebanner/bootstrap4.html (hunk 1 above)
Comment (tbentham) [reply cookiebanner/templates/cookiebanner/bootstrap4.html:76]: I tested it in Firefox with different scenarios and seemed fine. Which part of it are you not sure about?

**Comment by tbentham** [comment]
No worries! Thanks for getting back to me. I've replied to your review comments.

**Comment by andreasnuesslein** [comment]
@tbentham so it took me a while but here it finally is!: a new release.

the thing is, "bootstrap4" was always a template. i created a new default template `vanilla`, but i also added a section that describes how you can make your own template, easy peasy: https://github.com/sinnwerkstatt/django-cookiebanner#custom-template

**Pull request closed without merging**